include/configs/pistachio_bub.h

```diff
@@ -307,18 +304,25 @@
 
 #endif
 
+/*
+ * Openwrt have squashfs rootfs but legacy and other variant uses ubifs
+ * rootfs. To make all compatible dont pass root= for kernel booting
+ * from kernel ubi volume. For others, pass root= to correctly identify
+ * rootfs filesystem.
+ */
 #define NAND_BOOTCOMMAND	\
-	"setenv nandroot ubi.mtd=firmware$boot_partition $nandroot;"\
-	"setenv bootargs $console $earlycon $nandroot $bootextra $mtdparts panic=2;"\
 	"echo Attempting to boot from firmware$boot_partition;"\
+	"setenv ubimtd ubi.mtd=firmware$boot_partition;"\
 	"ubi part firmware$boot_partition || reset;"\
 	"if ubi check kernel; then "\
 		"echo Loading kernel from volume...;"\
+		"setenv bootargs $console $earlycon $ubimtd $bootextra $mtdparts panic=2;"\
 		"ubi read $loadaddr kernel || reset;"\
 	"else "\
 		"echo Loading kernel from rootfs...;"\
+		"setenv bootargs $console $earlycon $ubimtd $nandroot $bootextra $mtdparts panic=2; && "\
 		"ubifsmount ubi:rootfs && "\
-		"ubifsload $loadaddr $bootdir$fitfile || "NAND_BOOTCOMMAND_LEGACY"reset;"\
+		"ubifsload $loadaddr $bootdir$fitfile || "NAND_BOOTCOMMAND_LEGACY" "\
```

Comment:
The way i read is this:
Try ubifsload
if fail try nandboot
       in nandboot
       try booting (if success it will not return)
       else if failed to boot reset
If ubifsload succeed proceed with booting or if case of failure reset.
So booting is something which is not expected to return if success. only in case it fails we want to reset. This is reason reset look more to be part with actually command for which we want to test the condition?